avocado/core/mux.py

```diff
@@ -163,15 +164,25 @@ def initialize_mux(self, root, mux_path, debug):
         self.root = root
         self.mux_path = mux_path
         self.debug = debug
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
```

Comment:
I wonder how unreadable will be the variant name with big yaml files. I don't have a good solution though. Maybe we should truncate the variant id in a given number of characters. Then the hash would become useful, but a bigger part of it. Something like `foobarfoobar-nonononono-bigbranchname-foobarfoobar-nonononono-...-021911fb`